supervisor/agent/commander.go

```diff
@@ -0,0 +1,68 @@
+package agent
+
+import "context"
+
+// CommandConfig describes a systemd-based or an exec-based way of starting,
+// stopping and restarting the agent. Only one of `ExecConfig' or `SystemdConfig'
+// must be specified.
+type CommandConfig struct {
+	// Name is the path to the Agent executable. Path can be relative
+	// or absolute. A relative path should be relative to the CWD of the
+	// supervisor.
+	Name string `koanf:"name"`
+
+	// Args are the set of command-line arguments that must be specified along
+	// with the executable to start the Agent.
+	Args []string `koanf:"args"`
+}
+
+// Process represents information related to a running agent process.
+type Process struct {
+	// PID is the process id of the agent process that is currently running.
+	//
+	// A watchdog may use this PID to stop this agent process at a later time.
```

Comment:
Why is it necessary if there is already a Stop() method?

Reply:
Cover other reasons for agent termination?